File: leetcode202/leetcode202.py
class Solution:
    def isHappy(self, n: int) -> bool:
        def get_sum(n):
            _sum = 0
            while n > 0:
                _sum += (n % 10) ** 2
                n //= 10
            return  _sum
        
        set1 = set()
        while True:
            total_sum = get_sum(n)
            if total_sum == 1:
                return True
            if total_sum in set1:
                return False
            else:
                set1.add(total_sum)
            n = total_sum

File: leetcode202/leetcode202.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
    public:
        int getSum(int n){
            int sum = 0;
            while (n)
            {
                sum += (n % 10) * (n % 10);
                n /= 10;
            }
            return sum;
        }
    public:
        bool isHappy(int n) {
            unordered_set<int> set;
            while (true)
            {
                int sum = getSum(n);
                if (sum == 1) return true;
                if (set.find(sum) != set.end()) {
                    return false;
                }else{
                    set.insert(sum);
                }
                n = sum;
            }
        }
    };